### PR TITLE
Functional api for customized loss

### PR DESCRIPTION
**Pull request by xiaowang1105**
Ignore the Lambda layer cause it hasn't been merged to master yet.

### CODE REVIEW

File: pyzoo/zoo/examples/autograd/customloss_update.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+if __name__ == "__main__":
+    data_len = 1000
+    X_ = np.random.uniform(0, 1, (1000, 2))
+    Y_ = ((2 * X_).sum(1) + 0.4).reshape([data_len, 1])
+
+    a = Input(shape=(2,))
+    b = Dense(1)(a)
+    model = Model(input=a, output=b)
+
+    model.compile(optimizer=SGD(learningrate=1e-2),
```
Comment (hkvision) [inline pyzoo/zoo/examples/autograd/customloss_update.py]: remove metrics=None and validation_data=None if no validation is needed

File: pyzoo/zoo/examples/autograd/customloss_update.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+                  loss=mean_absolute_error,
+                  metrics=None)
+    model.fit(x=X_,
+              y=Y_,
+              batch_size=32,
+              nb_epoch=500,
+              validation_data=None,
+              distributed=False)
+
+    model.set_tensorboard('./log', 'customized layer and loss')
```
Comment (hkvision) [inline pyzoo/zoo/examples/autograd/customloss_update.py]: This should be called before `fit`

**Comment by hkvision** [comment]
Lambda has now been merged. Please update correspondingly.

**Comment by shane-huang** [comment]
>Can we use some meaningful data and a meaningful model to illustrate customized layer/loss?
Then can you find an example for our intern to refer to?

File: pyzoo/zoo/examples/autograd/customloss_update.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
+#
+# Copyright 2018 Analytics Zoo Authors.
```
Comment (zhichao-li) [inline pyzoo/zoo/examples/autograd/customloss_update.py]: How about rename this file to "custom.py" or any better name? @shane-huang

**Comment by zhichao-li** [comment]
Let's rebase the code and add `Lambda layer` into this example

File: pyzoo/zoo/examples/autograd/customloss_update.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    return x + 1.0
+
+if __name__ == "__main__":
+    data_len = 1000
+    X_ = np.random.uniform(0, 1, (1000, 2))
+    Y_ = ((2 * X_).sum(1) + 0.4).reshape([data_len, 1])
+
+    a = Input(shape=(2,))
+    b = Dense(1)(a)
+    c = Lambda(function=add_one_func)(b)
```
Comment (zhichao-li) [inline pyzoo/zoo/examples/autograd/customloss_update.py]: In respect of the API usage, I'm OK that add a such lamda layer here. Just not sure if we need to make it more meaningful. 
@jason-dai

**Comment by xiaowang1105** [comment]
jenkins passed http://172.168.2.101:8080/view/ZOO-PR-PIPELINE/job/ZOO-PR-Validation/254/
